### PR TITLE
Arm64 packages

### PR DESCRIPTION
**Pull request by hdabrowski**
### Desired Outcome

The outcome of this PR is to enable choosing architecture of output package (either amd64 or arm64).
By default it is set to amd64 in order to achieve backwards compatibility.
Apart from that, I have resolved some of the warning highlighted in RubyMine.

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue ID: [insert issue ID]

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: lib/conjur/debify.rb [hunk 1]
```diff
@@ -333,43 +339,67 @@ def copy_packages_from_container(container, package_name, dev_package_name)
       file_type = cmd_options[:output] || DEFAULT_FILE_TYPE
       container_cmd_options << "--file-type=#{file_type}"
 
+      # Set the architecture of the package
+      architecture = cmd_options[:architecture] || AMD64_ARCHITECTURE_NAME
+      if architecture != AMD64_ARCHITECTURE_NAME and architecture != ARM64_ARCHITECTURE_NAME
```
Comment (micahlee) [inline lib/conjur/debify.rb]: Can we use `&&` instead of `and` here?

File: lib/conjur/fpm/package.sh [hunk 2]
```diff
@@ -22,6 +22,10 @@ for i in "$@"; do
     file_type="${i#*=}"
     shift
     ;;
+  -fa=* | --architecture=*)
```
Comment (micahlee) [inline lib/conjur/fpm/package.sh]: Is the "f" in "-fa" significant here? If not, I might suggest only having the long form and not a short form. I'm not sure seeing "-fa" provides a reasonable affordance that this abbreviates "architecture".

File: lib/conjur/debify.rb [hunk 3]
```diff
@@ -33,24 +33,24 @@
 module DebugMixin
   DEBUG = ENV['DEBUG'].nil? ? true : ENV['DEBUG'].downcase == 'true'
 
-  def debug *a
+  def debug(* a)
```
Comment (micahlee) [inline lib/conjur/debify.rb:36]: Thanks for bringing this up to the current style standards! 😃

File: lib/conjur/debify.rb [hunk 4]
```diff
@@ -860,33 +860,11 @@ def build_test_image(appliance_image_id, project_name, packages)
...
-  # Use skips_post before a command to skip this
-  # block on that command only
-end
-
-on_error do |exception|
-  # Error logic here
-  # return false to skip default error handling
-  true
-end
+end
```
Comment (micahlee) [inline lib/conjur/debify.rb]: Mind restoring the final newline here?

**Review by micahlee — COMMENTED** [review]
This looks good! I only have a couple of small comments/suggestions. Let me know what you think!

File: lib/conjur/fpm/package.sh (hunk 2 above)
Comment (hdabrowski) [reply lib/conjur/fpm/package.sh]: Good point, i refactored it to the full flag only

**Review by micahlee — APPROVED** [review]
Thanks for this, @hdabrowski! LGTM

**Pull request merged**